### DISCUSSION TITLE
Converting ARM to adftools deployment

### DISCUSSION THREAD

**Discussion by sauravrout** [Azure-Player/azure.datafactory.tools#167] [Q&A] [answered]
Not an issue, but more a question.

From quite sometime I am using ARM way of deploying ADF, now I want to switch to adftools. I am not sure how my csv file will look like. My current prod substitution looks like this for ARM. Not sure if it can be converted to your csv format. There are some global parameters, which we replace in each environment. 

<img width="993" alt="image" src="https://user-images.githubusercontent.com/17978639/139266449-8e4370c6-5d76-4b56-a7bc-65452f15f8c6.png">

**Comment by NowinskiK** [comment, marked as answer]
Let me move this issue to the **Discussions**.

**Reply by NowinskiK** [reply]
It can be converted to the CSV config file, although carefully. 
This is the format the app expects:
```
type,name,path,value
linkedService,LS_AzureKeyVault,typeProperties.baseUrl,"https://kv-blog-uat.vault.azure.net/"
pipeline,PL_CopyMovies_with_param,parameters.DstBlobContainer.defaultValue,UAT
```
In ArmTemplate file(s), all first three arguments are concatenated with "_" separator.
Furthermore, except `dataFactory` they have no type of object in it, assuming the whole 'path' is unique.

So, in your case, a few examples should look as follow:
```
type,name,path,value
factory,"XXXXX-prod-adf",$.properties.globalParameters.gv_enableADF.value,"true"
trigger,parquetTrigger,$.properties.typeProperties.scope,"xxxxxx"
linkedService,0FT_KV,typeProperties.baseUrl,"https://xx-xxx-xxxxxx.vault.azure.net/"
```

Data Factory's location and name must be provided as input parameters.

**Reply by sauravrout** [reply]
thanks @NowinskiK. I will try to do it.

**Comment by sauravrout** [comment]
Hey @NowinskiK, I gave it a try. However I am getting this error and can't really figured out where/why this error comes up. Could you help. I am wondering if it is not treating _factory_ as a keyword.

![image](https://user-images.githubusercontent.com/17978639/144261209-78bfbcfd-ed40-4417-92c6-399d4f2c75bf.png)

This is how my csv file looks. If you look at my first question in this thread, I had posted my original ARM template overrides. Do  you think my csv file contains enough info ?

![image](https://user-images.githubusercontent.com/17978639/144220681-0d05abb9-e833-4b2b-9668-2f27d3847b6a.png)

**Reply by NowinskiK** [reply]
I think you have 2 files in `factory` folder which is wrong.

**Reply by sauravrout** [reply]
That doesn't help either. 

![image](https://user-images.githubusercontent.com/17978639/144347007-8106df94-2544-4a2d-aff9-7979204f2c06.png)

![image](https://user-images.githubusercontent.com/17978639/144347042-defbf16b-88b2-4ab0-992d-ae9d3316538c.png)

**Comment by sauravrout** [comment]
@NowinskiK Ok, it took me sometime to understand, but I finally got it. I think multiple files inside _factory_ folder is needed. It needs those file at least to find and replace global parameter values. The snippet you provided above needed some tweak. I had to replace last _ with "." in _path_.

`factory,"XXXXX-prod-adf",$.properties.globalParameters.gv_enableADF_value,"true"`

Thanks for your help.

Saurav
